src/yb/util/net/net_util.cc

```diff
@@ -140,24 +140,41 @@ Status HostPort::RemoveAndGetHostPortList(
   return Status::OK();
 }
 
+
+// Accepts entries like: ::1, 127.0.0.1, [::1]:7100, 0.0.0.0:7100, f.q.d.n:7100
 Status HostPort::ParseString(const string& str, uint16_t default_port) {
-  std::pair<string, string> p = strings::Split(str, strings::delimiter::Limit(":", 1));
+  uint32_t port;
 
-  // Strip any whitespace from the host.
-  StripWhiteSpace(&p.first);
+  // We look for the last colon and see if there is a valid port behind it
+  // If it looks like a valid port and there is only one colon, it's host:port
+  // If it looks like a valid port and the string starts with [ and there's a ]
+  // before the last colon, it's an IPv6 address [host]:port
+  // otherwise, just a host.
+  size_t pos = str.rfind(':');
+  if (pos != string::npos && pos > 1 && pos + 1 < str.length()
+      && SimpleAtoi(str.substr(pos + 1), &port)
+      && ((str[0] == '[' && str[pos - 1] == ']')
+          || count(str.begin(), str.end(), ':') == 1)) {
+
+    if (port == 0 || port > numeric_limits<uint16_t>::max()) {
+      return STATUS(InvalidArgument, "Invalid port", str);
+    }
 
-  // Parse the port.
-  uint32_t port;
-  if (p.second.empty() && strcount(str, ':') == 0) {
-    // No port specified.
-    port = default_port;
-  } else if (!SimpleAtoi(p.second, &port) ||
-             port > 65535) {
-    return STATUS(InvalidArgument, "Invalid port", str);
+    // Brackets indicate the address for IPv6 urls
+    if (str[0] == '[' && str[pos - 1] == ']') {
+      host_ = str.substr(1, pos - 2);
+      port_ = port;
+      return Status::OK();
+    } else {
+      host_ = str.substr(0, pos);
+      port_ = port;
+      return Status::OK();
+    }
   }
 
-  host_.swap(p.first);
-  port_ = port;
+  // Not in host:port or [host]:port format, assume it's a host
+  host_ = str;
```

Comment:
Should we remove `[]` from the host if present?

Reply:
At this point it sohuld not be present. If it was, it would have been caught by the [ ]:port if-statement above.

Reply:
Unless it was given without a port, I guess.. then we *could* strip it, but it's not strictly neccesary